HerPublicWebsite.BusinessLogic/Services/RegularJobs/PolicyTeamUpdateService.cs

```diff
@@ -48,15 +52,32 @@ IEmailSender emailSender
         return csvFileCreator.CreateReferralRequestOverviewFileData(newReferrals);
     }
 
-    private async Task<MemoryStream> BuildRecentReferralRequestFollowUpFileData (){
+    private async Task<(DateTime, DateTime)> RecentReferralRequestTimePeriod()
+    {
         DateTime endDate = await workingDayHelperService.AddWorkingDaysToDateTime(DateTime.Now, -10); 
-        DateTime startDate = await workingDayHelperService.AddWorkingDaysToDateTime(DateTime.Now.AddDays(-7), -10); 
+        DateTime startDate = await workingDayHelperService.AddWorkingDaysToDateTime(DateTime.Now.AddDays(-7), -10);
+        return (endDate, startDate);
+    }
+
+    private async Task<MemoryStream> BuildRecentLocalAuthorityReferralRequestFollowUpFileData (){
+        (DateTime endDate, DateTime startDate) = await RecentReferralRequestTimePeriod();
+        var newReferrals = await dataProvider.GetReferralRequestsBetweenDates(startDate, endDate);
+        return csvFileCreator.CreateLocalAuthorityReferralRequestFollowUpFileData(newReferrals);
+    }
+    private async Task<MemoryStream> BuildRecentConsortiumReferralRequestFollowUpFileData ()
+    {
+        (DateTime endDate, DateTime startDate) = await RecentReferralRequestTimePeriod();
         var newReferrals = await dataProvider.GetReferralRequestsBetweenDates(startDate, endDate);
-        return csvFileCreator.CreateReferralRequestFollowUpFileData(newReferrals);
+        return csvFileCreator.CreateConsortiumReferralRequestFollowUpFileData(newReferrals);
     }
 
-    private async Task<MemoryStream> BuildHistoricReferralRequestFollowUpFileData (){
+    private async Task<MemoryStream> BuildHistoricLocalAuthorityReferralRequestFollowUpFileData (){
+        var newReferrals = await dataProvider.GetAllReferralRequests();
+        return csvFileCreator.CreateLocalAuthorityReferralRequestFollowUpFileData(newReferrals);
+    }
+    
+    private async Task<MemoryStream> BuildHistoricConsortiumReferralRequestFollowUpFileData (){
         var newReferrals = await dataProvider.GetAllReferralRequests();
-        return csvFileCreator.CreateReferralRequestFollowUpFileData(newReferrals);
+        return csvFileCreator.CreateConsortiumReferralRequestFollowUpFileData(newReferrals);
```

Comment:
I think it would be good to combine the two (LA/consortium) methods in each case, so we only run the database query once in each case. It's unlikely to make much difference with the amounts of data we're dealing with, but often in situations like this the DB queries can end up being the bottleneck and halving* the time we spend doing that is always going to be good.

(*probably not, strictly speaking, but that's a whole other topic)

We're calling `ToListAsync` inside the `dataProvider` methods so we're not doing anything fancy with memory efficiency here anyway (which would be one reason not to do this - but again with the amounts of data we wouldn't realistically need to worry about that).

Reply:
Have touched on a couple of concepts here which aren't really relevant to the task at hand, but I'm happy to chat about them separately if you're interested